docs/pubsub-usage.rst

```diff
@@ -272,3 +272,30 @@ Fetch messages for a pull subscription without blocking (none pending):
    >>> messages = [recv[1] for recv in received]
    >>> [message.message_id for message in messages]
    []
+
+Fetch pending messages, acknowledging those whose processing doesn't raise an
+error:
+
+.. doctest::
+
+   >>> from gcloud import pubsub
+   >>> client = pubsub.Client()
+   >>> topic = client.topic('topic_name')
+   >>> subscription = topic.subscription('subscription_name')
+   >>> with topic.batch() as batch:
+   ...     batch.publish('this is the first message_payload')
+   ...     batch.publish('this is the second message_payload',
+   ...                   attr1='value1', attr2='value2')
+   >>> from gcloud.pubsub.subscription import AutoAck
+   >>> with AutoAck(subscription, max_messages=10) as ack:
+   ...     for ack_id, message in ack.items():
+   ...        try:
+   ...            do_something_with(message)
+   ...        except:
+   ...            del ack[ack_id]
```

Comment:
Yep, that is the strategy as @tmatsuo outlined it (he didn't spell it as dict-like, but the net effect was the same).